library/src/main/java/com/datatorrent/lib/io/fs/AbstractSingleFileOutputOperator.java

```diff
@@ -39,10 +41,37 @@
   @NotNull
   protected String outputFileName;
 
+  /**
+   * partitionedFileName string format specifier 
+      e.g. fileName_physicalPartionId -> %s_%d 
+   */
+  private String partitionedFileNameformat = "%s_%d";
+
+  /**
+   * Derived name for file based on physicalPartitionId
+   */
+  private transient String partitionedFileName;
+
+  /**
+   * Physical partition id for the current partition.
+   */
+  private transient int physicalPartitionId;
+
+  /**
+   * Initializing current partition id, partitionedFileName etc. {@inheritDoc}
+   */
+  @Override
+  public void setup(OperatorContext context)
```

Comment:
The **user** knows whether there are multiple partitions or just one, we don't need to figure it out. In the latter case, they need to have a way of turning off this behavior (e.g. can set the format string to null). There is no harm in doing this if it is properly documented.